GPU/Common/VertexDecoderCommon.cpp

```diff
@@ -496,41 +496,42 @@ void VertexDecoder::Step_ColorInvalid() const
 
 void VertexDecoder::Step_Color565() const
 {
-	u8 *c = decoded_ + decFmt.c0off;
+	u32 *c = (u32*)(decoded_ + decFmt.c0off);
 	u16 cdata = *(u16_le *)(ptr_ + coloff);
-	c[0] = Convert5To8(cdata & 0x1f);
-	c[1] = Convert6To8((cdata >> 5) & 0x3f);
-	c[2] = Convert5To8((cdata >> 11) & 0x1f);
-	c[3] = 255;
+	*c = Convert5To8(cdata & 0x1f);
+	*c |= Convert6To8((cdata >> 5) & 0x3f) << 8;
+	*c |= Convert5To8((cdata >> 11) & 0x1f) << 16;
+	*c |= 255 << 24;
 	// Always full alpha.
 }
 
 void VertexDecoder::Step_Color5551() const
 {
-	u8 *c = decoded_ + decFmt.c0off;
+	u32 *c = (u32*)(decoded_ + decFmt.c0off);
 	u16 cdata = *(u16_le *)(ptr_ + coloff);
 	gstate_c.vertexFullAlpha = gstate_c.vertexFullAlpha && (cdata >> 15) != 0;
-	c[0] = Convert5To8(cdata & 0x1f);
-	c[1] = Convert5To8((cdata >> 5) & 0x1f);
-	c[2] = Convert5To8((cdata >> 10) & 0x1f);
-	c[3] = (cdata >> 15) ? 255 : 0;
+	*c = Convert5To8(cdata & 0x1f);
+	*c |= Convert5To8((cdata >> 5) & 0x1f) << 8;
+	*c |= Convert5To8((cdata >> 10) & 0x1f) << 16;
+	*c |= (cdata >> 15) ? 255 << 24 : 0;
 }
 
 void VertexDecoder::Step_Color4444() const
 {
-	u8 *c = decoded_ + decFmt.c0off;
+	u32 *c = (u32*)(decoded_ + decFmt.c0off);
 	u16 cdata = *(u16_le *)(ptr_ + coloff);
 	gstate_c.vertexFullAlpha = gstate_c.vertexFullAlpha && (cdata >> 12) == 0xF;
+	*c = 0;
 	for (int j = 0; j < 4; j++)
-		c[j] = Convert4To8((cdata >> (j * 4)) & 0xF);
+		*c |= Convert4To8((cdata >> (j * 4)) & 0xF) << (j * 8);
```

Comment:
It might be better to use a temporary.  I think we use this sometimes to write to uncached GPU memory buffer space.  Also for Step_Color5551 / Step_Color565.  That said, hopefully we're using jit for these anyway...

-[Unknown]